server/src/main/java/io/druid/client/CachingClusteredClient.java

```diff
@@ -600,7 +749,7 @@ private void addSequencesFromServer(
               .withQuerySegmentSpec(segmentsOfServerSpec),
           responseContext
       );
-      final Function<T, Object> cacheFn = strategy.prepareForSegmentLevelCache();
+      final Function<T, Object> cacheFn = strategy.prepareForSegmentLevelCache()::apply;
```

Comment:
Could refactor to return the right `Function` type

Reply:
Yes, that would be nice, but I wanted to minimize the impact of this PR.